src/OpenTelemetry/Logs/OpenTelemetryLogger.cs

```diff
@@ -40,31 +41,74 @@ internal OpenTelemetryLogger(string categoryName, OpenTelemetryLoggerProvider pr
 
         public void Log<TState>(LogLevel logLevel, EventId eventId, TState state, Exception exception, Func<TState, Exception, string> formatter)
         {
-            if (!this.IsEnabled(logLevel)
-                || Sdk.SuppressInstrumentation)
+            if (!this.IsEnabled(logLevel) || Sdk.SuppressInstrumentation)
             {
                 return;
             }
 
             var processor = this.provider.Processor;
             if (processor != null)
             {
-                var options = this.provider.Options;
-
-                var record = new LogRecord(
-                    options.IncludeScopes ? this.ScopeProvider : null,
-                    DateTime.UtcNow,
-                    this.categoryName,
-                    logLevel,
-                    eventId,
-                    options.IncludeFormattedMessage ? formatter?.Invoke(state, exception) : null,
-                    options.ParseStateValues ? null : state,
-                    exception,
-                    options.ParseStateValues ? this.ParseState(state) : null);
-
-                processor.OnEnd(record);
-
-                record.ScopeProvider = null;
+                var provider = this.provider;
+                var options = provider.Options;
+
+                var record = new LogRecord
+                {
+                    Timestamp = DateTime.UtcNow,
+                    CategoryName = this.categoryName,
+                    LogLevel = logLevel,
+                    EventId = eventId,
+                    Exception = exception,
+                };
+
+                var activity = Activity.Current;
+                if (activity != null)
+                {
+                    record.TraceId = activity.TraceId;
+                    record.SpanId = activity.SpanId;
+                    record.TraceState = activity.TraceStateString;
+                    record.TraceFlags = activity.ActivityTraceFlags;
+                }
```

Comment:
If this logic is or was to be used elsewhere, I'd say it could be a function of `LogRecord`

Reply:
Yeah, something like `LogRecord.Clear()` or `LogRecord.Restock()`?